src/models/transactions/index.ts

```diff
@@ -69,57 +74,72 @@ export const saveTransaction = async (chainId: number, transaction: Transaction)
  * Store batch transactions and return nothing
  * @param transactions
  */
-export const saveTransactions = async (chainId: number, transactions: Transaction[]) => {
+export const saveTransactions = async (
+  chainId: number,
+  transactions: Transaction[],
+  receipts?: TransactionReceipt[]
+) => {
   if (_.isEmpty(transactions)) {
     return;
   }
 
-  const columns = new pgp.helpers.ColumnSet(
-    [
-      "hash",
-      "from",
-      "to",
-      "value",
-      "data",
-      "block_number",
-      "block_timestamp",
-      "gas_price",
-      "gas_used",
-      "gas_fee",
-    ],
-    { table: `transactions_${chainId}` }
-  );
+  const shouldParseReceipts = Array.isArray(receipts) && receipts.length;
 
-  const transactionsValues = _.map(transactions, (transaction) => ({
-    hash: toBuffer(transaction.hash),
-    from: toBuffer(transaction.from),
-    to: toBuffer(transaction.to),
-    value: transaction.value,
-    data: toBuffer(transaction.data),
-    block_number: transaction.blockNumber,
-    block_timestamp: transaction.blockTimestamp,
-    gas_price: transaction.gasPrice,
-    gas_used: transaction.gasUsed,
-    gas_fee: transaction.gasFee,
-  }));
+  const columns = [
+    "hash",
+    "from",
+    "to",
+    "value",
+    "data",
+    "block_number",
+    "block_timestamp",
+    "gas_price",
+    "gas_used",
+    "gas_fee",
+  ];
 
-  await idb.none(
-    `
-      INSERT INTO transactions_${chainId} (
-        hash,
-        "from",
-        "to",
-        value,
-        data,
-        block_number,
-        block_timestamp,
-        gas_price,
-        gas_used,
-        gas_fee
-      ) VALUES ${pgp.helpers.values(transactionsValues, columns)}
+  const receiptBasedColumns = ["nonce", "status"];
+
+  if (shouldParseReceipts) {
+    columns.push(...receiptBasedColumns);
+  }
+
+  const columnset = new pgp.helpers.ColumnSet(columns, { table: `transactions_${chainId}` });
+
+  const transactionsValues = _.map(transactions, (transaction, index) => {
+    const txnObject: Record<string, any> = {
+      hash: toBuffer(transaction.hash),
+      from: toBuffer(transaction.from),
+      to: toBuffer(transaction.to),
+      value: transaction.value,
+      data: toBuffer(transaction.data),
+      block_number: transaction.blockNumber,
+      block_timestamp: transaction.blockTimestamp,
+      gas_price: transaction.gasPrice,
+      gas_used: transaction.gasUsed,
+      gas_fee: transaction.gasFee,
+      nonce: transaction.nonce,
+      status: 1,
```

Comment:
if hardcoding status then better not to give

Reply:
Removed this line